authors/apps/comments/renderers.py

```diff
@@ -0,0 +1,22 @@
+import json
+from rest_framework.renderers import JSONRenderer
+from rest_framework.utils.serializer_helpers import ReturnDict, ReturnList
+
+
+class CommentJSONRenderer(JSONRenderer):
+    charset = 'utf-8'
+
+    def render(self, data, media_type=None, renderer_context=None):
+        # If the view throws an error (such as the user can't be authenticated
+        # or something similar), `data` will contain an `errors` key. We want
+        # the default JSONRenderer to handle rendering errors, so we need to
+        # check for this case.
+
+        if type(data) != ReturnList:  # try returning dict
+            errors = data.get('errors', None)
+            if errors is not None:
+                # As mentioned about, we will let the default JSONRenderer handle
+                # rendering errors.
+                return super(CommentJSONRenderer, self).render(data)
+        # Finally, we can render our data under the "comment" namespace.
+        return json.dumps({'comment': data})
```

Comment:
Thanks for the good work @wasibani-roy , was looking at the response we send back.

<img width="1418" alt="54521538-2c543800-497c-11e9-9b85-6a6ece713086" src="https://user-images.githubusercontent.com/20795487/54586480-39316400-4a2e-11e9-914a-d961e3728170.png">
I think saying comment['comment'] or comment['message'] in order to get comment data is confusing.


i think we could change the key to something like response or result,
so they so ```response.message,
response.comment```
to get what we send back.

Reply:
Please explain more about this i do not understand
Also remember there is a renderer at work there

Reply:
@wasibani-roy, I understand there is a renderer class and so in the renderer class, you could return `comment` when the length of `data` is one and `comments` when it's more as the key.